File: TechModule - Programming Fundamentals/08.ObjectsAndClassesEXERCISES/05.BookLibrary/BookLibrary.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;

namespace BookLibrary
{
    class BookLibrary
    {
        static void Main(string[] args)
        {
            int n = int.Parse(Console.ReadLine());
            List<Book> books = new List<Book>();
            for (int i = 0; i < n; i++)
            {
                string[] tokens = Console.ReadLine().Split();

                string title = tokens[0];
                string author = tokens[1];
                string publisher = tokens[2];
                DateTime releaseDate = DateTime.ParseExact(tokens[3], "dd.MM.yyyy", CultureInfo.InvariantCulture);
                string isbn = tokens[4];
                decimal price = decimal.Parse(tokens[5]);

                Book book = new Book(title, author, publisher, releaseDate, isbn, price);
                books.Add(book);
            }

            var authors = new Dictionary<string, decimal>();

            foreach (var book in books)
            {
                string authorName = book.Author;
                decimal price = book.Price;

                if (authors.ContainsKey(authorName) == false)
                {
                    authors.Add(authorName, price);
                }
                else
                {
                    authors[authorName] += price;
                }
            }

            Dictionary<string, decimal> sorted = authors.OrderByDescending(a => a.Value).ThenBy(a => a.Key).ToDictionary(x=>x.Key, x=> x.Value);

            foreach (var author in sorted)
            {
                Console.WriteLine($"{author.Key} -> {author.Value:f2}");
            }

        }
    }

    class Book
    {
        public Book(string title, string author, string publisher, DateTime releaseDate, string isbn, decimal price)
        {
            Title = title;
            Author = author;
            Publisher = publisher;
            ReleaseDate = releaseDate;
            Isbn = isbn;
            Price = price;
        }

        public string Title { get; set; }
        public string Author { get; set; }
        public string Publisher { get; set; }
        public DateTime ReleaseDate { get; set; }
        public string Isbn { get; set; }
        public decimal Price { get; set; }
    }
}

File: Software Technologies/09.JavaBasics/src/BookLibrary.java
import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.util.ArrayList;
import java.util.Scanner;
import java.util.TreeMap;

public class BookLibrary {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        scanner.nextLine();

        ArrayList<Book> books = new ArrayList<>();
        Library library = new Library();
        for (int i = 0; i < n; i++) {
            String[] input = scanner.nextLine().split(" ");
            Book book = new Book();
            book.setTitle(input[0]);
            book.setAuthor(input[1]);
            book.setPublisher(input[2]);
            book.setReleaseDate(LocalDate.parse(input[3], DateTimeFormatter.ofPattern("dd.MM.yyyy")));
            book.setIsbn(input[4]);
            book.setPrice(Double.parseDouble(input[5]));
            books.add(book);
        }
        library.setName("My Library");
        library.setBookList(books);
        TreeMap<String, Double> authorPrice = new TreeMap<>();
        for (Book book : library.getBookList()) {
            if (!authorPrice.containsKey(book.getAuthor()))
            {
                authorPrice.put(book.getAuthor(), book.getPrice());
            }
            else
            {
                double oldValue = authorPrice.get(book.getAuthor());
                authorPrice.put(book.getAuthor(), oldValue + book.getPrice());
            }
        }
        authorPrice.entrySet().stream().sorted((e1, e2) -> {
            int compareResult = Double.compare(e2.getValue(), e1.getValue());
            if (compareResult == 0){
                compareResult = e1.getKey().compareTo(e2.getKey());
            }
            return compareResult;
        }).forEach(e-> System.out.printf("%s -> %.2f%n", e.getKey(), e.getValue()));
    }

    public static class Library {
        private String name;
        private ArrayList<Book> bookList;

        public Library() {
        }

        public Library(String name, ArrayList<Book> bookList) {
            this.name = name;
            this.bookList = bookList;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public ArrayList<Book> getBookList() {
            return bookList;
        }

        public void setBookList(ArrayList<Book> bookList) {
            this.bookList = bookList;
        }
    }

    public static class Book {
        private String title;
        private String author;
        private String publisher;
        private LocalDate releaseDate;
        private String isbn;
        private Double price;

        public Book() {
        }

        public Book(String title, String author, String publisher, LocalDate releaseDate, String isbn, Double price) {
            this.title = title;
            this.author = author;
            this.publisher = publisher;
            this.releaseDate = releaseDate;
            this.isbn = isbn;
            this.price = price;
        }

        public String getTitle() {
            return title;
        }

        public void setTitle(String title) {
            this.title = title;
        }

        public String getAuthor() {
            return author;
        }

        public void setAuthor(String author) {
            this.author = author;
        }

        public String getPublisher() {
            return publisher;
        }

        public void setPublisher(String publisher) {
            this.publisher = publisher;
        }

        public LocalDate getReleaseDate() {
            return releaseDate;
        }

        public void setReleaseDate(LocalDate releaseDate) {
            this.releaseDate = releaseDate;
        }

        public String getIsbn() {
            return isbn;
        }

        public void setIsbn(String isbn) {
            this.isbn = isbn;
        }

        public Double getPrice() {
            return price;
        }

        public void setPrice(Double price) {
            this.price = price;
        }
    }
}